spring-integration-rsocket/src/main/java/org/springframework/integration/rsocket/outbound/RSocketOutboundGateway.java

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.rsocket.outbound;
+
+import java.util.function.Consumer;
+
+import org.reactivestreams.Publisher;
+
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.Expression;
+import org.springframework.integration.expression.ExpressionUtils;
+import org.springframework.integration.expression.ValueExpression;
+import org.springframework.integration.handler.AbstractReplyProducingMessageHandler;
+import org.springframework.lang.Nullable;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.rsocket.RSocketRequester;
+import org.springframework.messaging.rsocket.RSocketStrategies;
+import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
+import org.springframework.util.MimeType;
+import org.springframework.util.MimeTypeUtils;
+
+import io.rsocket.RSocket;
+import io.rsocket.RSocketFactory;
+import io.rsocket.transport.ClientTransport;
+import reactor.core.publisher.Mono;
+
+/**
+ * An Outbound Messaging Gateway for RSocket client requests.
+ *
+ * @author Artem Bilan
+ *
+ * @since 5.2
+ *
+ * @see RSocketRequester
+ */
+public class RSocketOutboundGateway extends AbstractReplyProducingMessageHandler {
```

Comment:
Shouldn't this be `RSocketMessageHandler` since it supports `fireAndForget` (adapter semantics)?

Reply:
I thought about that, too, but there is already a `RSocketMessageHandler` in Spring Messaging and its point is about a *server* side, so we don't have a choice and choose a different name.

Re. `fireAndForget` (a-la one-way): it is really doesn't fit to reactive model. We always need to return something (in this case a `Mono<Void>`) to let end-user to decide when to subscribe to it and, essentially, perform a request.
So, even if it is empty, it still going to be a reply from this gateway. The `FluxMessageChannel` downstream will subscribe to that `Mono` or `async` mode will do an eager subscription otherwise.

In summary: we always return something from here, even if it is just about `fireAndForget`

Reply:
ok